File: backend/src/main/java/com/technicalescaperoom/backend/controller/player/StorylineController.java
package com.technicalescaperoom.backend.controller.player;

import com.technicalescaperoom.backend.config.security.PlayerPrincipal;
import com.technicalescaperoom.backend.dto.player.StorylineDto;
import com.technicalescaperoom.backend.service.StorylineService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/player/game")
@RequiredArgsConstructor
public class StorylineController {

    private final StorylineService storylineService;

    @GetMapping("/story")
    public ResponseEntity<StorylineDto> getStoryline(@AuthenticationPrincipal PlayerPrincipal principal) {
        StorylineDto storyline = storylineService.getStorylineForPlayer(principal);
        return ResponseEntity.ok(storyline);
    }
}

File: backend/controllers/player/storylineController.js
const storylineService = require('../../services/storylineService');

class StorylineController {
  async getStoryline(req, res, next) {
    try {
      const storyline = await storylineService.getStorylineForPlayer(req.player);
      res.json(storyline);
    } catch (err) {
      next(err);
    }
  }
}

module.exports = new StorylineController();
